scripts/tile_benchmark/benchmark.py

```diff
@@ -0,0 +1,365 @@
+    for host in run_params.keys():
+        if host == 'atlas':
+            print "Getting images from atlas"
+            elapsed_time = get_data(
+                "atlas",
+                'atlas hardcoded img',
+                int(run_params[host][1]),
+                int(run_params[host][2]))
+        elif host == 'adrc':
+            print "Getting images from adrc"
+            elapsed_time = get_data(
+                "adrc",
+                'adrc hardcoded img',
+                int(run_params[host][1]),
+                int(run_params[host][2]))
+        else:
+            print"Getting images from girder"
+            elapsed_time = get_data(
+                run_params[host][0],
+                run_params[host][1],
+                int(run_params[host][2]),
+                int(run_params[host][3]))
+        run_params[host].append(elapsed_time)
+
+    plot_data_and_save(run_params)
+
+
+def get_data(host, image_id, requested_levels, requested_num_tiles):
+    """get request information from each server an request tiles"""
+    num_levels = -1
+    data = {}
+    # get data about server, host url and params
+    if host == "atlas":
+
+        data = atlas_init()
+    elif host == "adrc":
+
+        data = adrc_init()
+    else:
+        data = girder_init(host, image_id)
+    num_levels = data[NUM_LEVELS]
+    # if successful begin to grab tiles from each level
+    if num_levels != -1:
+        if requested_levels > num_levels:
+            requested_levels = num_levels
+        # get elapsed time info numpy array include
+        elapsed_time_arr, count = \
+            iterate_over_tiles(
+                host,
+                data,
+                requested_levels,
+                requested_num_tiles)
+
+        # create a new array of correct size and plot
+
+        return elapsed_time_arr
+
+
+def iterate_over_tiles(host, data, requested_levels, requested_num_tiles):
+    """iterate from level 0 to higher levels requesting tiles"""
+    total_tiles = 0
+    for z in range(requested_levels):
+        total_tiles += 2 ** (z * 2)
+
+    count = 0
+    elapsed_time_arr = np.zeros([total_tiles], dtype=float)
+    # collect from all levels
+    for z in range(requested_levels):
+        print"started level %d" % (z)
+
+        for x in range(2 ** z):
+            for y in range(2 ** z):
+
+                if host == "atlas":
+
+                    request_time = atlas_request(data, z, x, y)
+                elif host == "adrc":
+
+                    request_time = adrc_request(data, z, x, y)
+                else:
+
+                    request_time = girder_request(data, z, x, y)
+                if request_time != -1:
+                    elapsed_time_arr[count] = request_time
+                    count += 1
+                    if count == requested_num_tiles:
+                        elapsed_arr_final = np.zeros([count], dtype=float)
+                        np.copyto(elapsed_arr_final, elapsed_time_arr[:count])
```

Comment:
You can use [`numpy.trim_zeros`](http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.trim_zeros.html#numpy.trim_zeros) or [`numpy.resize`](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.resize.html) here.